test/uk/gov/hmrc/gform/models/FormModelSupport.scala

```diff
@@ -100,13 +100,13 @@ trait FormModelSupport extends GraphSpec {
       FormComponentIdToFileIdMapping.empty
     )
 
-  def mkFormModelOptics(
+  def mkFormModelOptics[D <: DataOrigin](
     formTemplate: FormTemplate,
     data: VariadicFormData[SourceOrigin.OutOfDate]
-  )(implicit messages: Messages, lang: LangADT): FormModelOptics[DataOrigin.Browser] = {
+  )(implicit messages: Messages, lang: LangADT): FormModelOptics[D] = {
     val authCache: AuthCacheWithForm = mkAuthCacheWithForm(formTemplate)
     FormModelOptics
-      .mkFormModelOptics[DataOrigin.Browser, Id, SectionSelectorType.Normal](data, authCache, recalculation)
+      .mkFormModelOptics[D, Id, SectionSelectorType.Normal](data, authCache, recalculation)
```

Comment:
Can we revert this changes in `mkFormModelOptics`. Going abstract means loosing useful information, and I don't think is good price to pay for test sake.

Use `isInstanceOf` in the test code if needed.

Reply:
Makes sense, corrected